app/helpers/auth_helper.rb

```diff
@@ -27,7 +27,9 @@ def authorize_role!
     when :user
       params = request.filtered_parameters
       if params["controller"] == "locks" &&
-          ["create", "update", "lock_from_pool"].include?(params["action"])
+          ["create", "update", "lock_from_pool"].include?(params["action"]) ||
+         params["controller"] == "pools" &&
```

Comment:
Totally agreed. Added separate method to just disable pool, will modify update method so API could be used to update any info with master credentials..